src/emmy/expression/compile.cljc

```diff
@@ -532,9 +353,9 @@
   - `state-model`: a sequence of variables representing the structure of the
```

Comment:
I'm quite happy that this seems to be unifying the different compile functions, between the compile-state-fn and compile-fn.

Do you think it makes sense to have a mode that compiles something that looks like a state derivative into a function of the parameters that returns a function of the local tuple, instead of flattening both levels into one function? Not for this PR, but I wanted to throw that into the design soup.